src/core/sse2/vector.rs

```diff
@@ -261,6 +261,11 @@ impl Vector<f32> for __m128 {
     fn max(self, other: Self) -> Self {
         unsafe { _mm_max_ps(self, other) }
     }
+
+    #[inline(always)]
+    fn clamp(self, min: Self, max: Self) -> Self {
+        self.max(min).min(max)
```

Comment:
This version should also have a `glam_assert` for consistency, e.g. `glam_assert!(min.cmp_le(max).all())`

Reply:
Hmm, the assert actually complicates things a bit it turns out. For the clamp there's no Vector3/Vector4 implementation required because we just ignore lanes we don't care about, but for the assert we do care about the lanes, or rather we don't want say the w lane to factor into the Vec3 clamp assert. You could remove it for now and I'll deal with it later. Adding `Vector3` and `Vector4` implementations, instead of `Vector` might fix it.